llmfoundry/models/layers/attention.py

```diff
@@ -462,29 +464,72 @@ def __init__(
             self.softmax_scale = 1 / math.sqrt(self.d_model / self.n_heads)
         self.attn_dropout_p = attn_pdrop
 
-        if self.reuse_kv_layer_idx is None:
-            self.Wqkv = build_fc(
-                name=fc_type_name,
-                in_features=self.d_model,
-                out_features=self.d_model + 2 * self.kv_n_heads * self.head_dim,
-                fc_kwargs=fc_type,
-            )
-            # for param init fn; enables shape based init of fused layers
-            fuse_splits = [
-                i * self.head_dim
-                for i in range(1, self.n_heads + 2 * self.kv_n_heads)
-            ]
-            self.Wqkv._fused = (0, fuse_splits)
+        if self.fused_weights:
```

Comment:
I think this code can be simplified a bit, currently a bunch of duplication. basically `if reuse_kv_layer_idx`, we're always making Wq, and then if fused we make wqkv, else we make wk and wv

Reply:
yea wanted to get a branch up for quick testing. sg.

Reply:
done